os/arch/arm/src/armv7-a/arm_cpuidlestack.c

```diff
@@ -141,6 +141,7 @@ int up_cpu_idlestack(int cpu, struct tcb_s *tcb, size_t stack_size)
   tcb->adj_stack_size  = SMP_STACK_SIZE;
   tcb->stack_alloc_ptr = (void *)stack_alloc;
   tcb->stack_base_ptr  = tcb->stack_alloc_ptr;
+  tcb->adj_stack_ptr = (void *)(stack_alloc + SMP_STACK_SIZE - 4);
```

Comment:
Don't we need to do this for other TCBs?

Reply:
For other tasks, it will be taken care when createstack() API is called. The issue is only with idle stack.